### PR TITLE
refactor: only use shards

### LINKED ISSUE #5277: chore: update all tests that use default pubsubTopic to use shard based pubsubTopic
# Problem

Named sharding is being deprecated in nwaku and also the fleets that support named shards are being shut-down. Any tests that depend on the default pubsubTopic i.e `/waku/2/default-waku/proto` should be migrated to use sharding based pubsubTopic. 

Especially if the tests are dependent on fleets, then they would start failing. 

# Implementation

Need to analyze which tests use this and modify them accordingly. 
Note that wakunode would have to be started with clusterID `16` in the tests when this change is being made.
https://github.com/status-im/status-go/pull/4665/ can be referred as i had modified some of waku tests to switch to shard based pubsubTopics.

# Acceptance Criteria

All tests should pass successfully.

cc @richard-ramos

### PR DESCRIPTION
**Pull request by chaitanyaprem**
Closes #5277

### CODE REVIEW

**Review by kaichaosun — APPROVED** [review]

File: protocol/messenger_communities.go [hunk 1]
```diff
@@ -2614,6 +2602,26 @@ func (m *Messenger) SetCommunityShard(request *requests.SetCommunityShard) (*Mes
...
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
+	fmt.Println("%%%%%%%%%%%")
```
Comment (igor-sirotin) [inline protocol/messenger_communities.go]: leftover?

Comment (igor-sirotin) [inline protocol/messenger_communities.go]: is this still needed?

**Review by igor-sirotin — APPROVED** [review]
🙌

File: protocol/messenger_communities.go (hunk 1 above)
Comment (richard-ramos) [reply protocol/messenger_communities.go]: oops

Comment (richard-ramos) [reply protocol/messenger_communities.go]: The more %%%% the better! :kek:

**Comment by richard-ramos** [comment]
For the scenarios that are failing, I think we're sending the messages on the wrong pubsub topic, i.e. 
```
rawMessage := common.RawMessage{
		Payload:             payload,
		CommunityID:         community.ID(),
		SkipEncryptionLayer: true,
		MessageType:         protobuf.ApplicationMetadataMessage_COMMUNITY_EDIT_SHARED_ADDRESSES,
		PubsubTopic:         community.PubsubTopic(), // TODO: confirm if it should be sent in community pubsub topic
		ResendType:          common.ResendTypeRawMessage,
	}
```
I think this should have probably been sent on `shard.DefaultNonProtectedPubsubTopic()` (If I change that `TestMessengerCommunitiesTokenPermissionsSuite/TestEditSharedAddresses` passes).

I'm investigating the other cases.

**Comment by richard-ramos** [comment]
@jrainville, @mprakhov
Please do check the latest commit. I did change the pubsub topic of some messages.

The question is:
Can these messages be sent before the user has joined a community? If so, they should be send/received in either the default non protected topic or in the default shard, and not in the community shard. Pls let me know.

**Comment by jrainville** [comment]
> Can these messages be sent before the user has joined a community? If so, they should be send/received in either the default non protected topic or in the default shard, and not in the community shard. Pls let me know.

@richard-ramos both of these can only be called when the user has already joined the community.

**Comment by richard-ramos** [comment]
I'll revert last commit then

**Comment by richard-ramos** [comment]
'kay, all green now :)

**Review by richard-ramos — APPROVED** [review]

**Comment by igor-sirotin** [comment]
Manual testing probably is good idea for this one?
cc @anastasiyaig @churik

**Comment by chaitanyaprem** [comment]
> Manual testing probably is good idea for this one? cc @anastasiyaig @churik

Yep, was going to create sttus-desktop PR with this. 
But both of them seem to be OOO , not sure who will test.

**Comment by chaitanyaprem** [comment]
> Manual testing probably is good idea for this one? cc @anastasiyaig @churik

Here is the status-desktop PR https://github.com/status-im/status-desktop/pull/15455

**Comment by igor-sirotin** [comment]
> > Manual testing probably is good idea for this one? cc @anastasiyaig @churik
> 
> Yep, was going to create sttus-desktop PR with this. But both of them seem to be OOO , not sure who will test.

Let's ping @glitchminer and @qoqobolo 🙂

**Comment by pavloburykh** [comment]
> > Manual testing probably is good idea for this one? cc @anastasiyaig @churik
> 
> Yep, was going to create sttus-desktop PR with this. But both of them seem to be OOO , not sure who will test.

@chaitanyaprem thank you for the PR and pinging QA team. Could you please rebase status go branch to the latest develop and create status mobile PR as well? 

Also, would appreciate some details on what exactly has been affected and how should we test this PR from user perspective (what test scenarios/areas)? Thank you.

**Comment by chaitanyaprem** [comment]
> > > Manual testing probably is good idea for this one? cc @anastasiyaig @churik
> > 
> > 
> > Yep, was going to create sttus-desktop PR with this. But both of them seem to be OOO , not sure who will test.
> 
> @chaitanyaprem thank you for the PR and pinging QA team. Could you please rebase status go branch to the latest develop and create status mobile PR as well?
> 
> Also, would appreciate some details on what exactly has been affected and how should we test this PR from user perspective (what test scenarios/areas)? Thank you.

Done and here it is https://github.com/status-im/status-mobile/pull/20658. 
Not sure if many scenarios can be tested from mobile though, because they involve creating commuities and assigning shards to them. Maybe mobile can be used as a device to test send/receive messages and joini communities in this case.

**Comment by pavloburykh** [comment]
@chaitanyaprem I have left a comment in [mobile PR](https://github.com/status-im/status-mobile/pull/20658#issuecomment-2210363516), posting it here as well, so everyone is aware about the problem. 


> Unfortunately we cannot proceed with testing as currently we experience global Waku issues which result in communities failed to be fetched. This is a reason why [e2e](https://github.com/status-im/status-mobile/pull/20658#issuecomment-2210300356) run in this PR has mostly failed. 

> I have already pinged Waku team about the problem, waiting for the fix. 

> Here is example of logs of a user from this PR who failed to fetch community when following community link. 

[Status-debug-logs - 2024-07-05T101140.382.zip](https://github.com/user-attachments/files/16106612/Status-debug-logs.-.2024-07-05T101140.382.zip)

**Comment by pavloburykh** [comment]
Hey @cammellos! What do you think, is this a priority to test this PR on mobile or we can rely on e2e results from our side? Note that our e2e do not cover sharded communities.

**Comment by cammellos** [comment]
@pavloburykh thanks, given the priorities for 2.30, I don't think we have enough capacity in the QA team to test something that's not strictly needed for 2.30. If @chaitanyaprem thinks that this is not impacting anything unless the community is manually sharded, then e2e tests is probably good enough, but if it impacts production code, best to leave it for later

**Comment by chaitanyaprem** [comment]
> @pavloburykh thanks, given the priorities for 2.30, I don't think we have enough capacity in the QA team to test something that's not strictly needed for 2.30. If @chaitanyaprem thinks that this is not impacting anything unless the community is manually sharded, then e2e tests is probably good enough, but if it impacts production code, best to leave it for later

afaik it doesn't affect any other code apart from sharded communities.
Also, i have tested by creating a community and assigning it a shard 128 via RPC (which requires a privateKey) to be specified and i was able to fetch the community from both mobile and desktop.
Maybe there is a bug in the UI for assigning shard which is causing the issue.

 But i'll want a yes from the great @richard-ramos as well to be sure.

**Comment by chaitanyaprem** [comment]
> > @pavloburykh thanks, given the priorities for 2.30, I don't think we have enough capacity in the QA team to test something that's not strictly needed for 2.30. If @chaitanyaprem thinks that this is not impacting anything unless the community is manually sharded, then e2e tests is probably good enough, but if it impacts production code, best to leave it for later
> 
> afaik it doesn't affect any other code apart from sharded communities. But i'll want a yes from the great @richard-ramos as well to be sure.

**Comment by richard-ramos** [comment]
IMO it should be fine to merge this PR. The normal behavior of the app is maintained, and I think that as long as the custom shard RPC methods still work fine, which as far as we know, it is the case, we should be good to go!

**Pull request merged**